clmm/support/mock_data.py

```diff
@@ -542,7 +567,10 @@ def _draw_source_redshifts(zsrc, zsrc_min, zsrc_max, ngals):
     else:
         raise ValueError(f"zsrc must be a float, chang13 or desc_srd. You set: {zsrc}")
 
-    return GCData([zsrc_list, zsrc_list], names=("ztrue", "z"))
```

Comment:
Why not just adding `coordinate_system` to `_draw_source_redshifts` instead of creating a new  function (`_create_gcdata`) ?